kms/kms_test.go

```diff
@@ -540,12 +546,26 @@ func TestImportEndToEnd(t *testing.T) {
 	}
 	cryptoKeyVersionName := fmt.Sprintf("%s/cryptoKeyVersions/1", cryptoKeyName)
 
-	// Check the state of the imported key.
-	if err := checkStateImportedKey(&b, cryptoKeyVersionName); err != nil {
-		t.Fatal(err)
-	}
-	if got, want := b.String(), "Current state"; !strings.Contains(got, want) {
-		t.Errorf("checkStateImportedKey: expected %q to contain %q", got, want)
+	// Wait for the key to finish importing.
+	importInProgress := true
+	importInProgressStatus := kmspb.CryptoKeyVersion_CryptoKeyVersionState_name[int32(kmspb.CryptoKeyVersion_PENDING_IMPORT)]
+	for importInProgress {
+		b.Reset()
+		if err := checkStateImportedKey(&b, cryptoKeyVersionName); err != nil {
+			t.Fatal(err)
+		}
+
+		got := b.String()
+		if want := "Current state"; !strings.Contains(got, want) {
+			t.Errorf("checkStateImportedKey: expected %q to contain %q", got, want)
+		}
+
+		if !strings.Contains(got, importInProgressStatus) {
+			importInProgress = false
```

Comment:
You don't need both of the statements in this `if` block. I would remove the `importInProgress` variable altogether. Change the for loop so that it doesn't have a condition variable:

```
for {
    // ...
    if !strings.Contains(got, importInProgressStatus) {
        break
    }
}
```

Reply:
Thanks for the suggestion. Updated.